riscv/devices.h

```diff
@@ -79,6 +79,11 @@ class clint_t : public abstract_device_t {
 #define PLIC_MAX_DEVICES 1024
 
 struct plic_context_t {
+  plic_context_t(processor_t* proc, bool mmode)
+    : proc(proc), mmode(mmode), priority_threshold(0), enable{}, pending{},
```

Comment:
I think it would be preferable to use in-class member initializers for everything but `proc` and `mmode`. E.g:
```
uint8_t priority_threshold{};
uint32_t enable[PLIC_MAX_DEVICES/32]{};
```

Reply:
Agreed; I'm mentally stuck in C++03.

Reply:
> Is there any significance to the mix of parentheses and curly braces?

I was misremembering that `{}` was syntax for array initialization only.